packages/fe-fpm-writer/src/building-block/index.ts

```diff
@@ -0,0 +1,155 @@
+import { render } from 'ejs';
+import type { Editor } from 'mem-fs-editor';
+import { join } from 'path';
+import type { BuildingBlock } from './types';
+import { DOMParser, XMLSerializer } from '@xmldom/xmldom';
+import * as xpath from 'xpath';
+import format from 'xml-formatter';
+import { getErrorMessage, validateBasePath } from '../common/validate';
+
+/**
+ * Generates a building block into the provided xml view file.
+ *
+ * @param {string} basePath - the base path
+ * @param {string} viewOrFragmentPath - the path of the view or fragment xml file relative to the base path
+ * @param {string} aggregationPath - the aggregation xpath
+ * @param {BuildingBlock} buildingBlockData - the building block data
+ * @param {Editor} [fs] - the memfs editor instance
+ * @returns {Editor} the updated memfs editor instance
+ */
+export function generateBuildingBlock<T extends BuildingBlock>(
+    basePath: string,
+    viewOrFragmentPath: string,
+    aggregationPath: string,
+    buildingBlockData: T,
+    fs: Editor
+): Editor {
+    // Validate the base and view paths
+    validateBasePath(basePath, fs);
+    if (!fs.exists(join(basePath, viewOrFragmentPath))) {
+        throw new Error(`Invalid view path ${viewOrFragmentPath}.`);
+    }
+
+    // Read the view xml and template files and update contents of the view xml file
+    const viewDocument = getViewDocument(basePath, viewOrFragmentPath, fs);
+    const templateDocument = getTemplateDocument(buildingBlockData, viewDocument, fs);
+    fs = updateViewFile(basePath, viewOrFragmentPath, aggregationPath, viewDocument, templateDocument, fs);
+
+    return fs;
+}
+
+/**
+ * Returns the view xml file document.
+ *
+ * @param {string} basePath - the base path
+ * @param {string} viewPath - the path of the xml view relative to the base path
+ * @param {Editor} fs - the memfs editor instance
+ * @returns {Document} the view xml file document
+ */
+function getViewDocument(basePath: string, viewPath: string, fs: Editor): Document {
```

Comment:
The function also works with fragments, also indicated in your public function, so I'd suggest to rename it as well.